astropy/io/misc/hdf5.py

```diff
@@ -269,6 +271,13 @@ def write_table_hdf5(table, output, path=None, compression=False,
         group, name = None, path
 
     if isinstance(output, (h5py.File, h5py.Group)):
+        if len(list(output.keys())) > 0 and name == '__astropy__':
+            raise ValueError("table path should always be set via the "
+                             "path= argument when writing to existing "
+                             "files")
+        elif name == '__astropy__':
+            warnings.warn("table path was not set via the path= argument; "
+                          "using default path {}".format(path))
```

Comment:
Small nitpick: if I do ``path='__astropy__'`` I'll still get this message, so maybe leave it set to ``None`` until this point and then set it to ``__astropy__``?

Reply:
`path` has to be a string already at line 268. Maybe mangle even more the string (e.g. `__astropy__table__dataset__extnsn__` :) )?
The recursive nature of this function already creates challenges when one wants to debug it. Setting more control variables to None/bools and propagate them in the sub-calls would create even more complications. This is why I opted for a mangled string instead.